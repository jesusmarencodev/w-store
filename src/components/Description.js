import React from 'react';

const Description = ({ description }) => {
  return (
    <div className="product__description">
      <div className="shadow"></div>
      <p> {description}</p>
      <div className="shadow"></div>
    </div>
  );
};

export default Description;
