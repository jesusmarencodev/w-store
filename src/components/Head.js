import React from 'react';

import { useAmount } from '../hooks/useAmount';
import { useShoppingCar } from '../hooks/useShoppingCar';
import { useSelector } from 'react-redux';

const Head = () => {
  const { amount } = useAmount();
  const { shoppingCar } = useSelector((state) => state.products);
  const { handlerShowCar } = useShoppingCar();

  return (
    <>
      <header className="head d-flex a-center">
        <div className="head__logo">
          <img className="head__img" src="store.png" alt="store" />
        </div>
        <div
          className={`head__shop d-flex ${shoppingCar && 'head__shop--active'}`}
        >
          <div
            onClick={() => handlerShowCar(true)}
            className={`${
              shoppingCar ? 'shop__button--active' : 'shop__button'
            }`}
          >
            <img className="head__img-car" src="car.png" alt="store" />
            <span className={`shop-total ${shoppingCar && 'shop-active'}`}>
              ${amount}
            </span>
          </div>
          {shoppingCar && (
            <button
              onClick={() => handlerShowCar(false)}
              className="head__shop-button"
            >
              X
            </button>
          )}
        </div>
      </header>
    </>
  );
};

export default Head;
