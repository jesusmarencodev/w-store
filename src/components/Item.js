import React from 'react';

const Item = ({ item }) => {
  return (
    <div className="shoppingCar__item">
      <div className="item d-flex a-center">
        <div className="item__quantity">{item.quantity}</div>
        <img src={item.image} alt={item.name} />
      </div>
      <div className="shadow"></div>
    </div>
  );
};

export default Item;
