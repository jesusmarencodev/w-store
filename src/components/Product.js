import React from 'react';
import { useDispatch } from 'react-redux';
import { getSelected } from '../redux/feature/productReducer';

export const Product = ({ product }) => {
  const dispatch = useDispatch();

  const handlerProduct = ({ id }) => {
    dispatch(getSelected(id));
  };

  return (
    <>
      <div
        onClick={() => handlerProduct(product)}
        className={`product d-flex a-center j-center  ${
          product.special ? ' product--large' : ' product--normal'
        }`}
      >
        <div className="product__quantity d-flex a-center j-center">
          <span>{product.quantity}</span>
        </div>
        <img
          className={`product__img${
            product.special ? ' product__img--large d-flex' : ''
          }`}
          src={product.image}
          alt={product.name}
        ></img>
      </div>
    </>
  );
};
