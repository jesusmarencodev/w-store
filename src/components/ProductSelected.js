import React from 'react';
import Description from './Description';
import { useProduct } from '../hooks/useProduct';

const ProductSelected = () => {
  const { product, handlerQuantity } = useProduct();
  return (
    <>
      {product?.name ? (
        <div>
          <div className={`product product-selected d-flex a-center j-center`}>
            <div className="product__quantity d-flex a-center j-center">
              <span>{product.quantity}</span>
            </div>
            <img
              className={`product__img d-flex`}
              src={product.image}
              alt={product.name}
            ></img>
          </div>
          <div className="product__buy d-flex a-center">
            <p className="buy__data">
              <span className="data-name">{product.name} </span>
              <span className="data-name data-point">&middot;</span>
              <span className="data-name data-price">${product.price}</span>
            </p>
            <div className="product__buttons">
              <button
                disabled={product.quantity === 0 ? true : false}
                onClick={() => handlerQuantity(-1)}
              >
                -
              </button>
              <button onClick={() => handlerQuantity(1)}>+</button>
            </div>
          </div>
          <Description description={product.description} />
        </div>
      ) : (
        <p className="product__description" style={{ paddingTop: '2.5rem' }}>
          Please choose a product on the left.
        </p>
      )}
    </>
  );
};

export default ProductSelected;
