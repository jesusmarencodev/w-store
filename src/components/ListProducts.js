import React from 'react';
import { useSelector } from 'react-redux';
import { Product } from './Product';

const ListProducts = () => {
  const { productList } = useSelector((state) => state.products);

  return (
    <article className="listProducts">
      <div className="shadow"></div>
      <div className="scrollbar">
        <div className="listProducts__list">
          {productList?.map((product) => (
            <Product key={product.id} product={product} />
          ))}
        </div>
      </div>
    </article>
  );
};

export default ListProducts;
