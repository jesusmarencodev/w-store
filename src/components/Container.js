import React from 'react';
import ListProducts from './ListProducts';
import QuantityProduct from './QuantityProduct';

export const Container = () => {
  return (
    <div className="d-flex container">
      <ListProducts />
      <QuantityProduct />
    </div>
  );
};
