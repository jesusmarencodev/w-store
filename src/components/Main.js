import React from 'react';
import Head from './Head';
import { Container } from './Container';

const Main = () => {
  return (
    <div className="main">
      <Head />
      <h3 className="main__title">Store</h3>
      <Container />
    </div>
  );
};

export default Main;
