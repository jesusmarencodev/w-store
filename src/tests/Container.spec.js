import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { Container } from '../components/Container';

const mockStore = configureStore([]);

describe('Container component', () => {
  it('renders correctly', () => {
    const store = mockStore({
      products: {
        shoppingCar: false,
        productList: [],
      },
    });

    const component = renderer.create(
      <Provider store={store}>
        <Container />
      </Provider>
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
