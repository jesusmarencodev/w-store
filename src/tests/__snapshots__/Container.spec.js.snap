// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container component renders correctly 1`] = `
<div
  className="d-flex container"
>
  <article
    className="listProducts"
  >
    <div
      className="shadow"
    />
    <div
      className="scrollbar"
    >
      <div
        className="listProducts__list"
      />
    </div>
  </article>
  <aside
    className="carShopping"
  >
    <div
      className="shadow"
      style={
        Object {
          "marginRight": "2.5rem",
        }
      }
    />
    <div
      className="scrollbar"
    >
      <div>
        <p
          className="product__description"
          style={
            Object {
              "paddingTop": "2.5rem",
            }
          }
        >
          Please choose a product on the left.
        </p>
      </div>
    </div>
  </aside>
</div>
`;
